DOC-734 Adds redpanda_common and redpanda inputs and outputs and redpanda config updates (#97)
redpanda-data/rp-connect-docs: modules/components/pages/inputs/redpanda.adoc

*Type*: `array`

```yml
# Examples

topics:
  - foo
  - bar

topics:
  - things.*

topics:
  - foo,bar

topics:
  - foo:0
  - bar:1
  - bar:3

topics:
  - foo:0,bar:1,bar:3

topics:
  - foo:0-5
```

=== `regexp_topics`

Whether listed topics are interpreted as regular expression patterns for matching multiple topics. When topics are specified with explicit partitions, this field must remain set to `false`.

*Type*: `bool`

*Default*: `false`

=== `rack_id`

A rack specifies where the client is physically located, and changes fetch requests to consume from the closest replica as opposed to the leader replica.

*Type*: `string`

*Default* `""`

=== `start_from_oldest`

Determines whether to consume from the oldest available offset. Otherwise, messages are consumed from the latest offset. This setting is applied when creating a new consumer group or the saved offset no longer exists.

*Type*: `bool`

*Default*: `true`

=== `fetch_max_bytes`

The maximum amount of bytes that a broker tries to send during a fetch.

If individual records are larger than the `fetch_max_bytes` value, brokers will still send them.

*Type*: `string`

*Default*: `50MiB`

=== `fetch_min_bytes`

> The minimum amount of bytes that a broker tries to send during a fetch. This field is equivalent to the Java setting `fetch.min.bytes`.

Reviewer: ```suggestion
The minimum number of bytes that a broker tries to send during a fetch. This field is equivalent to the Java setting `fetch.min.bytes`.
```

Author: Updated.

*Type*: `string`

*Default*: `1B`

=== `fetch_max_partition_bytes`

> The maximum amount of bytes that are consumed from a single partition in a fetch request. This field is equivalent to the Java setting `fetch.max.partition.bytes`.

Reviewer: ```suggestion
The maximum number of bytes that are consumed from a single partition in a fetch request. This field is equivalent to the Java setting `fetch.max.partition.bytes`.
```

= redpanda
// tag::single-source[]
:type: input
:page-beta: true
:categories: ["Services"]

component_type_dropdown::[]

Consumes topic data from one or more Kafka brokers.

ifndef::env-cloud[]
Introduced in version 4.39.0.
endif::[]

> Use this connector in conjunction with the xref:components:outputs/redpanda.adoc[`redpanda` output], if you need to move topic data between Redpanda clusters, or other Apache Kafka clusters.

Author: Is the accurate?

Reviewer 2: No, the redpanda output isn't necessary, they can both be used independently and with any other plugins.

> NOTE: If you need to read data from one or more topics on a Redpanda cluster, and then write the data to another topic on the same cluster, consider using the xref:components:inputs/redpanda_common.adoc[`redpanda_common` input] and xref:components:outputs/redpanda_common.adoc[output] instead.

Reviewer 2: ```suggestion
NOTE: If you intend to use multiple redpanda components that have a common set of credentials, consider using the xref:components:inputs/redpanda_common.adoc[`redpanda_common` input] and xref:components:outputs/redpanda_common.adoc[output] instead.
```

Author: Updated.

```

=== `sasl[].mechanism`

The SASL mechanism to use.

*Type*: `string`

|===
| Option | Summary

| `AWS_MSK_IAM`
| AWS IAM-based authentication as specified by the `aws-msk-iam-auth` Java library
| `OAUTHBEARER`
| OAuth Bearer-based authentication
| `PLAIN`
| Plain text authentication
| `SCRAM-SHA-256`
| SCRAM-based authentication as specified in RFC 5802
| `SCRAM-SHA-512`
| SCRAM-based authentication as specified in RFC 5802
| `none`
| Disable SASL authentication

|===

=== `sasl[].username`

A username for `PLAIN` or `SCRAM-*` authentication.

*Type*: `string`

*Default*: `""`

=== `sasl[].password`

A password for `PLAIN` or `SCRAM-*` authentication.

include::components:partial$secret_warning.adoc[]

*Type*: `string`

*Default*: `""`

=== `sasl[].token`

The token to use for a single session's `OAUTHBEARER` authentication.

*Type*: `string`

*Default*: `""`

=== `sasl[].extensions`

Key/value pairs to add to `OAUTHBEARER` authentication requests.

*Type*: `object`

=== `sasl[].aws`

AWS specific fields for when the `mechanism` is set to `AWS_MSK_IAM`.

*Type*: `object`

=== `sasl[].aws.region`

The AWS region to target.

*Type*: `string`

*Default*: `""`

> === `sasl[].aws.endpoint`

Author: Are all AWS fields configurable in Cloud?

Reviewer 2: Yes

The maximum amount of bytes that are consumed from a single partition in a fetch request. This field is equivalent to the Java setting `fetch.max.partition.bytes`.

If a single batch is larger than the `fetch_max_partition_bytes` value, the batch is still sent so that the client can make progress. 

*Type*: `string`

*Default*: `1MiB`

=== `consumer_group`

An optional consumer group. When this value is specified: 

- The partitions of any topics, specified in the `topics` field, are automatically distributed across consumers sharing a consumer group
- Partition offsets are automatically committed and resumed under this name

Consumer groups are not supported when you specify explicit partitions to consume from in the `topics` field.

*Type*: `string`

=== `commit_period`

The period of time between each commit of the current partition offsets. Offsets are always committed during shutdown.

*Type*: `string`

*Default*: `5s`

=== `partition_buffer_bytes`

A buffer size (in bytes) for each consumed partition, which allows the internal queuing of records before they are flushed. Increasing this value may improve throughput but results in higher memory utilization. 

> Each buffer can grow slightly beyond this value.

Author: When we say slightly, how much?

Reviewer 2: I believe it depends on the broker configuration, so it's probably best to stay vague on the details.
